_posts/2016-02-27-write-code-for-the-dumbest-person-you-know.md

```diff
@@ -0,0 +1,71 @@
+
+I probably missed something but I have to say that [I started to be pissed off](https://twitter.com/Selrahcd/status/701794483726323712) again - I think there is some kind of pattern here. I tried to understand, because, after all, you're clever. Why would you do something that complicated when just using the class from the library, or your wrapper, would have done the job ? Maybe you were protecting us from the coupling to that library and making its use simpler. I have to confess that I was surprised when I saw a massive chain of getters going all the way down to get the library object you were wrapping and pass it to the some other object.
+
+This are some of the times I noticed I was dumb. Hard truth...
+
+Thank you for beeing an eye opener.
+
+
+## I used to be clever though
+
+I used to be clever though. I used to write some pretty pieces of code which were doing all kind of cool stuff. I was theorically able to change the way the whole system was working by changing one parameter in a configuraion file. My code was able to cope with all future issues I had foreseen. I knew how to do some tricky things with all sort of tools.
+
+But time goes on and aging probably made me dumb.
+
+Aging or discoveries and failures.
+
+Abstractions are good when you know what you're abstracting. Apparently you only know what you're abstracting based on the information you currently have. I couldn't predict future and the next needs. I tried though, I failed, and ended up [locked down by my clever abstraction](http://www.sandimetz.com/blog/2016/1/20/the-wrong-abstraction). And as we all do I started to cheat, broke my abstraction, used getters in order to get the internals and be able to manipulate them.
+
+Not once but several times.
+
+I made complicated stuff too. They looked hard to understand at first glance but once you managed to wrap your head around the problem being solved everyone could see how my solution was elegant and was able to get it. I was at least. I wrote it after all. Then I moved on, worked on something else, and had to came by to it later probably because of something to fix. It seems like complicated code tend to be buggy. Worst, apparently time had increased code complexity. I was now unable to understand easily what was going on with this code and all choices I made a few days before.
+
+I have seen people reluctant to use some of my code. In my opinion it was not that hard to understand but you probably needed someone to guide you the first time you used the library because it was magic in some kind of way. We had a solution, we probably should have written documentation. Or simpler code : [something the junior of your team can grasp by himself](http://www.infoq.com/presentations/8-lines-code-refactoring).
```

Comment:
but **they** probably needed someone to guide **them** the first time **they** used